src/format/LemLibOdomFormatV0_1.tsx

```diff
@@ -0,0 +1,230 @@
+
+// observable class
+class GeneralConfigImpl implements GeneralConfig {
+  @IsPositive()
+  @Expose()
+  robotWidth: number = 12;
+  @IsPositive()
+  @Expose()
+  robotHeight: number = 12;
+  @IsBoolean()
+  @Expose()
+  robotIsHolonomic: boolean = false;
+  @IsBoolean()
+  @Expose()
+  showRobot: boolean = false;
+  @ValidateNumber(num => num > 0 && num <= 1000) // Don't use IsEnum
+  @Expose()
+  uol: UnitOfLength = UnitOfLength.Inch;
+  @IsPositive()
+  @Expose()
+  pointDensity: number = 2; // inches
+  @IsPositive()
+  @Expose()
+  controlMagnetDistance: number = 5 / 2.54;
+  @Type(() => FieldImageSignatureAndOrigin)
+  @ValidateNested()
+  @IsObject()
+  @Expose()
+  fieldImage: FieldImageSignatureAndOrigin<FieldImageOriginType> =
+    getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+
+  @Exclude()
+  private format_: LemLibOdomFormatV0_1;
+
+  constructor(format: LemLibOdomFormatV0_1) {
+    this.format_ = format;
+    makeAutoObservable(this);
+
+    reaction(
+      () => this.uol,
+      action((_: UnitOfLength, oldUOL: UnitOfLength) => {
+        convertGeneralConfigUOL(this, oldUOL);
+      })
+    );
+
+    intercept(this, "fieldImage", change => {
+      const { assetManager } = getAppStores();
+
+      if (assetManager.getAssetBySignature(change.newValue.signature) === undefined) {
+        change.newValue = getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+      }
+
+      return change;
+    });
+  }
+
+  get format() {
+    return this.format_;
+  }
+
+  getConfigPanel() {
+    return <></>;
+  }
+}
+
+// observable class
+class PathConfigImpl implements PathConfig {
+  @ValidateNumberRange(-Infinity, Infinity)
+  @Expose()
+  speedLimit: NumberRange = {
+    minLimit: { value: 0, label: "" },
+    maxLimit: { value: 0, label: "" },
+    step: 0,
+    from: 0,
+    to: 0
+  };
+  @ValidateNumberRange(-Infinity, Infinity)
+  @Expose()
+  bentRateApplicableRange: NumberRange = {
+    minLimit: { value: 0, label: "" },
+    maxLimit: { value: 0, label: "" },
+    step: 0,
+    from: 0,
+    to: 0
+  };
+  @ValidateNumber(num => num >= 0.1 && num <= 255)
+  @Expose()
+  maxDecelerationRate: number = 127;
+
+  @Exclude()
+  readonly format: LemLibOdomFormatV0_1;
+
+  @Exclude()
+  public path!: Path;
+
+  constructor(format: LemLibOdomFormatV0_1) {
+    this.format = format;
+    makeAutoObservable(this);
+  }
+
+  getConfigPanel() {
+    return (<></>);
+  }
+}
+
+// observable class
+export class LemLibOdomFormatV0_1 implements Format {
+  isInit: boolean = false;
+  uid: string;
+
+  private gc = new GeneralConfigImpl(this);
+  private readonly events = new Map<keyof HistoryEventMap<CancellableCommand>, Set<Function>>();
+
+  constructor() {
+    this.uid = makeId(10);
+    makeAutoObservable(this);
+  }
+
+  createNewInstance(): Format {
+    return new LemLibOdomFormatV0_1();
+  }
+
+  getName(): string {
+    return "LemLib Odometry v0.1.x (inch coordinates)";
```

Comment:
Maybe rename it to `LemLib Odometry Code Generator v0.4.x`? We can have a discussion.